src/Strategy/BackedEnumStrategy.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Hydrator\Strategy;
+
+use BackedEnum;
+use Laminas\Hydrator\Exception\DomainException;
+use Laminas\Hydrator\Strategy\Exception\InvalidArgumentException;
+use ValueError;
+
+use function get_debug_type;
+use function is_a;
+use function is_scalar;
+use function sprintf;
+
+use const PHP_VERSION_ID;
+
+final class BackedEnumStrategy implements StrategyInterface
+{
+    private string $enumClass;
+
+    /**
+     * @param class-string $enumClass
```

Comment:
Is there a concept such as `enum-string`?

Reply:
Should perhaps be a `class-string<BackedEnum>`

Reply:
Good call